crates/bdk/src/wallet/mod.rs

```diff
@@ -427,27 +427,97 @@ impl<D> Wallet<D> {
             .next()
     }
 
-    /// Return a single transactions made and received by the wallet
-    ///
-    /// Optionally fill the [`TransactionDetails::transaction`] field with the raw transaction if
-    /// `include_raw` is `true`.
-    pub fn get_tx(&self, txid: Txid, include_raw: bool) -> Option<TransactionDetails> {
+    /// Returns a `TxOut` known by this wallet corresponding to `outpoint` if it exists in the
+    /// wallet's database whether spent or not.
+    pub fn get_txout(&self, outpoint: &OutPoint) -> Option<TxOut> {
+        self.indexed_graph
+            .graph()
+            .all_txouts()
+            .filter_map(|(op, txo)| {
+                if op.eq(outpoint) {
+                    Some(txo.clone())
+                } else {
+                    None
+                }
+            })
+            .next()
```

Comment:
nit: you can use `find_map`
```
    pub fn get_txout(&self, outpoint: &OutPoint) -> Option<TxOut> {
        self.indexed_graph
            .graph()
            .all_txouts()
            .find_map(|(op, txo)| {
                if op.eq(outpoint) {
                    Some(txo.clone())
                } else {
                    None
                }
            })
    }
```

Reply:
Thanks for the tip! but while fixing it I found that there's already an `TxGraph::get_txout()` function so I'm using that instead.